### PR TITLE
Fix 183 EAP build regression 

### PR DESCRIPTION
**Pull request by wezhang**
There are following type issues solved:

1. More strict NPE checking for Kotlin codes.
   - Fix it by removing `?` from the type.
2. 183 EAP API generic type changes, such as: `RunConfigurationBase` -> `RunConfigurationBase<T>`
   - Leverage Java class to bypass the check. Java code ignore the generic type checking.
3. Some class names are changed in the new version Scala plugin, such as: `SbtSystemSettings` -> `SbtSettings`
   - Remove that dependency by invoking from `ExternalSystemManager` and `ProjectImportProvider` level API.

### CODE REVIEW

**Comment by Eskibear** [comment]
Thanks for fixing the `compileKotlin` task, I see it fails the java compilation now. 
```org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':compileJava'```

**Comment by wezhang** [comment]
Looks like I used a "new" API of 182, compiling failure of 181 version:

```
+ ./gradlew clean buildPlugin -s -Pintellij_version=IC-2018.1 -Pdep_plugins=org.intellij.scala:2018.1.8
Starting a Gradle Daemon (subsequent builds will be faster)
> Task :clean UP-TO-DATE
w: /jenkinshome/jenkins/workspace/Azure_Toolkits_for_Java/AzureToolsManager_PR/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStorageWithUploadPathPanel.kt: (91, 101): Unchecked cast: ComboBoxModel<Any!>! to DefaultComboBoxModel<String>
w: /jenkinshome/jenkins/workspace/Azure_Toolkits_for_Java/AzureToolsManager_PR/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/SparkSubmissionJobUploadStorageWithUploadPathPanel.kt: (122, 105): Unchecked cast: DefaultComboBoxModel<String> to DefaultComboBoxModel<Any>
> Task :compileKotlin

> Task :compileJava
warning: Supported source version 'RELEASE_7' from annotation processor 'com.microsoft.azure.management.apigeneration.LangDefinitionProcessor' less than -source '1.8'
/jenkinshome/jenkins/workspace/Azure_Toolkits_for_Java/AzureToolsManager_PR/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/SbtProjectGenerator.java:199: error: cannot find symbol
        final VirtualFile projectFile = VfsUtil.findFile(Paths.get(root + "/build.sbt"), true);
                                               ^
  symbol:   method findFile(Path,boolean)
  location: class VfsUtil
```

> Thanks for fixing the `compileKotlin` task, I see it fails the java compilation now.
> `org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':compileJava'`

**Comment by Eskibear** [comment]
Forget about the CI error log in this PR. I've merged your code to my 183 PR, it turns out as the following:
```
azure-toolkit-for-intellij\src\com\microsoft\azure\sparkserverless\serverexplore\ui\SparkServerlessProvisionDialog.java:149: error: cannot find symbol
        refreshButton.addActionListener(event -> ctrlProvider.updateAvailableAU().subscribe(
                                                             ^
  symbol:   method updateAvailableAU()
  location: variable ctrlProvider of type SparkServerlessClusterProvisionCtrlProvider
azure-toolkit-for-intellij\src\com\microsoft\azure\sparkserverless\serverexplore\ui\SparkServerlessProvisionDialog.java:181: error: cannot find symbol
                ctrlProvider.updateAvailableAUAndTotalAU()
                            ^
  symbol:   method updateAvailableAUAndTotalAU()
  location: variable ctrlProvider of type SparkServerlessClusterProvisionCtrlProvider
azure-toolkit-for-intellij\src\com\microsoft\azure\sparkserverless\serverexplore\ui\SparkServerlessProvisionDialog.java:227: error: cannot find symbol
            refreshButton.setEnabled(data.getRefreshEnabled());
                                         ^
  symbol:   method getRefreshEnabled()
  location: variable data of type SparkServerlessClusterProvisionSettingsModel
azure-toolkit-for-intellij\src\com\microsoft\azure\sparkserverless\serverexplore\ui\SparkServerlessProvisionDialog.java:255: error: cannot find symbol
                .setRefreshEnabled(refreshButton.isEnabled())
                ^
  symbol:   method setRefreshEnabled(boolean)
  location: class SparkServerlessClusterProvisionSettingsModel
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
4 errors
```

Can you help to take a look?

**Comment by wezhang** [comment]
You may need a rebase and a clean build of `Utils` 😃 

> Forget about the CI error log in this PR. I've merged your code to my 183 PR, it turns out as the following:
> 
> ```
> azure-toolkit-for-intellij\src\com\microsoft\azure\sparkserverless\serverexplore\ui\SparkServerlessProvisionDialog.java:149: error: cannot find symbol
>         refreshButton.addActionListener(event -> ctrlProvider.updateAvailableAU().subscribe(
>                                                              ^
>   symbol:   method updateAvailableAU()
>   location: variable ctrlProvider of type SparkServerlessClusterProvisionCtrlProvider
> azure-toolkit-for-intellij\src\com\microsoft\azure\sparkserverless\serverexplore\ui\SparkServerlessProvisionDialog.java:181: error: cannot find symbol
>                 ctrlProvider.updateAvailableAUAndTotalAU()
>                             ^
>   symbol:   method updateAvailableAUAndTotalAU()
>   location: variable ctrlProvider of type SparkServerlessClusterProvisionCtrlProvider
> azure-toolkit-for-intellij\src\com\microsoft\azure\sparkserverless\serverexplore\ui\SparkServerlessProvisionDialog.java:227: error: cannot find symbol
>             refreshButton.setEnabled(data.getRefreshEnabled());
>                                          ^
>   symbol:   method getRefreshEnabled()
>   location: variable data of type SparkServerlessClusterProvisionSettingsModel
> azure-toolkit-for-intellij\src\com\microsoft\azure\sparkserverless\serverexplore\ui\SparkServerlessProvisionDialog.java:255: error: cannot find symbol
>                 .setRefreshEnabled(refreshButton.isEnabled())
>                 ^
>   symbol:   method setRefreshEnabled(boolean)
>   location: class SparkServerlessClusterProvisionSettingsModel
> Note: Some input files use or override a deprecated API.
> Note: Recompile with -Xlint:deprecation for details.
> Note: Some input files use unchecked or unsafe operations.
> Note: Recompile with -Xlint:unchecked for details.
> 4 errors
> ```
> Can you help to take a look?

**Comment by Eskibear** [comment]
Thanks, it works.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/SbtProjectGenerator.java [hunk 1]
```diff
@@ -173,17 +176,38 @@ private String generateSbtFileContent() {
...
-        sbtProjectSettings.setExternalProjectPath(root);
+        if (projectImportProviders[0] == null) {
+            return;
+        }
 
-        SbtSystemSettings sbtSystemSettings = SbtSystemSettings.getInstance(project);
-        HashSet<SbtProjectSettings> projects = ContainerUtilRt.newHashSet(sbtSystemSettings.getLinkedProjectsSettings());
-        projects.add(sbtProjectSettings);
-        sbtSystemSettings.setLinkedProjectsSettings(projects);
+        final VirtualFile projectFile = VfsUtil.findFile(Paths.get(root + "/build.sbt"), true);
```
Comment (Eskibear) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/SbtProjectGenerator.java]: I'm not sure whether the path delimiter will cause issues in different platforms, but `Paths.get(root, "build.sbt")` looks better for me.

**Comment by Eskibear** [comment]
By merging commits from PR #2231 you can get the CI passed. We'are going to support 2018.2 & 2018.3 in the coming release if there's no significant regression in EAP 183.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/SbtProjectGenerator.java (hunk 1 above)
Comment (wezhang) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/SbtProjectGenerator.java]: Thanks! To fix it!

**Pull request merged**